src/components/sketches/Scene5_happy.ts

```diff
@@ -0,0 +1,290 @@
+import p5 from 'p5';
+import {
+  MAX_RING_PARTICLES,
+  EASING,
+  RING_RADIUS,
+  RING_COVERAGE,
+  EXPLODE_RANGE,
+  EXPLODE_OFFSET,
+  FIREWORK_LIFESPAN,
+  LIFESPAN_DECREMENT,
+  VELOCITY_CHANGE,
+  TRAIL_SIZE,
+} from '../../utils/constants';
+
+export default function sketch(p: p5): void {
+  let canvas: p5.Renderer;
+  const fireworks: Firework[] = [];
+  const ringParticles: ringParticle[] = [];
+  let gravity: p5.Vector;
+
+  p.setup = () => {
+    canvas = p.createCanvas(window.innerWidth, window.innerHeight);
+    //p.colorMode(p.HSB);
+    canvas.id('p5-background');
+    gravity = p.createVector(0, 0.095);
+    p.stroke(255);
+    p.strokeWeight(4);
+    p.background(0);
+
+    // Setting mode to degrees for spawning ringParticles in circle
+    p.angleMode(p.DEGREES);
+    p.background(0);
+    // Sets 0,0 to centre
+    p.translate(p.width / 2, p.height / 2);
+    // Max number of ringParticles
+    p.frameRate(30);
+  };
+
+  p.windowResized = () => {
+    p.resizeCanvas(window.innerWidth, window.innerHeight, true);
+  };
+
+  p.draw = () => {
+    p.background(0, 0, 0, 25); // easy trails but deletes rings quickly
+
+    p.ellipse(p.mouseX,p.mouseY, 5,5);
+    if (p.random(1) < 0.03) {
+      fireworks.push(new Firework());
+    }
+
+    for (let i = fireworks.length - 1; i >= 0; i--) {
+      fireworks[i].update();
+      fireworks[i].show();
+
+      if (fireworks[i].done()) {
+        fireworks.splice(i, 1);
+      }
+    }
+
+    // If too many ringParticles on screen
+    if (ringParticles.length > MAX_RING_PARTICLES) {
+      // Delete old ringParticles (from beginning of array)
+      ringParticles.splice(0, p.floor((ringParticles.length - MAX_RING_PARTICLES * 0.8) * EASING));
```

Comment:
is this complexity needed to delete particles? why cant we just do like 2 or smth?

Reply:
like why do we need to do `p.floor((ringParticles.length - MAX_RING_PARTICLES * 0.8) * EASING)`?

why cant we just remove it by a constant amount